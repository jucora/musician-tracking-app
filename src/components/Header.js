import React from 'react';

const Header = () => (
  <header>
    <h1>Musician Tracking App</h1>
    <div className="title-subtitle">Keep track of your musical skills!</div>
  </header>
);

export default Header;
