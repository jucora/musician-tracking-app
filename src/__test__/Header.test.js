import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../components/Header';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});

it('should have a title', () => {
  const { getByTestId } = render(<Header text="Musician Tracking App" />);
  expect(getByTestId('headerTitle')).toHaveTextContent('Musician Tracking App');
});

it('should have a subtitle', () => {
  const { getByTestId } = render(
    <Header text="Keep track of your musical skills!" />,
  );
  expect(getByTestId('title-subtitle')).toHaveTextContent(
    'Keep track of your musical skills!',
  );
});

test('header div child should have a class title-subtitle', () => {
  const { getByText } = render(<Header />);
  expect(getByText('Keep track of your musical skills!')).toHaveClass(
    'title-subtitle',
  );
});
