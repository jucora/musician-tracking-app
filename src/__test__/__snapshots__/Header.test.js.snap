// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h1
      data-testid="headerTitle"
    >
      Musician Tracking App
    </h1>
    <div
      class="title-subtitle"
      data-testid="title-subtitle"
    >
      Keep track of your musical skills!
    </div>
  </header>
</DocumentFragment>
`;
